src/libraries/System.IO.FileSystem/tests/Base/BaseGetSetTimes.cs

```diff
@@ -70,6 +72,77 @@ public void SettingUpdatesProperties()
             });
         }
 
+        [Fact]
+        [PlatformSpecific(~TestPlatforms.Browser)]
+        public void SettingUpdatesPropertiesAfterAnother()
+        {
+            T item = GetExistingItem();
+
+            // These linq calls make an IEnumerable of pairs of functions that are not identical
+            // (eg. not (creationtime, creationtime)), includes both orders as seperate entries
+            // as they it have different behavior in reverse order (of functions), in addition
+            // to the pairs of functions, there is a reverse bool that allows a test for both
+            // increasing and decreasing timestamps as to not limit the test unnecessarily.
+            // Only testing with utc because it would be hard to check if lastwrite utc was the
+            // same type of method as lastwrite local since their .Getter fields are different.
+            var timeFunctionsUtc = TimeFunctions(requiresRoundtripping: true).Where((f) => f.Kind == DateTimeKind.Utc);
+            var booleanArray = new bool[] { false, true };
+            Assert.All(timeFunctionsUtc.SelectMany((x) => timeFunctionsUtc.SelectMany((y) => booleanArray.Select((reverse) => (x, y, reverse)))).Where((fs) => fs.x.Getter != fs.y.Getter), (functions) =>
```

Comment:
Hi, I'm not exactly sure how to write a `Theory`, but I will look into it. Also, did you read the comment? It is important to have all of these variants and I am not personally sure how to replicate them without using the already existing `TimeFunctions` method.

Reply:
Also, the reason I added this test is because it was actually failing with earlier versions of the code that I committed.

Reply:
Or I could put the order in a comment above to the test implementation assuming all 3 time functions are available.

Reply:
Like this (assuming these are the correct orders):
![image](https://user-images.githubusercontent.com/25316035/118392601-ca87d980-b67d-11eb-9aff-fe91ad61c52b.png)

Reply:
My 2c, it would be clearer broken out a [InlineData]. It's not as compact, but it's easier to read at a glance, and tests are modified much less than product code so it makes sense to optimize for readability.

Here's some examples 
https://github.com/dotnet/runtime/blob/e7b743cae325d4720fceaf7f84f68aaaf7bb7dbe/src/libraries/System.IO.FileSystem/tests/Base/FileGetSetAttributes.cs#L30-L39
https://github.com/dotnet/runtime/blob/e7b743cae325d4720fceaf7f84f68aaaf7bb7dbe/src/libraries/System.IO.FileSystem/tests/FileStream/LockUnlock.cs#L121-L130

Reply:
I could do that, but I'd have to change the class thing to have a new field which would identify it (eg. string field with text like "creation_utc"). If you guys think that's an okay change to make, I'm happy to make it.

Reply:
@danmoseley like this.
![image](https://user-images.githubusercontent.com/25316035/118536406-0eb6cf00-b78f-11eb-817a-4773760861f4.png)

Reply:
@danmoseley, please check the latest commit to see if its what you meant.

Reply:
Ah, I see. Yes, hard coding the function names is rather ugly. What about simply a nested loop over TimeFunctions? ie.,
```c#
foreach (TimeFunction tf1 in TimeFunctions())
{
    foreach (TimeFunction tf2 in TimeFunctions())
   {
        foreach (bool reverse in new bool[] { true, false })
        {
                  ...
   }
}
```
I didn't look carefully: would something like that work?
Unlike a `[Theory]` each combination would not be run and reported individually, but maybe that's less important.

Reply:
@danmoseley that would be almost alright, except for:
- You wouldn't be able to tell which one failed if it failed since `Assert.All` provides an index, whereas this doesn't
- You missed the `requiresRoundtripping: true`
- You haven't excluded repeat functions which would fail the test, the easiest way I could think of to do this is what I was doing:
```cs
var timeFunctionsUtc = TimeFunctions(requiresRoundtripping: true).Where((f) => f.Kind == DateTimeKind.Utc)
//and then
timeFunctionsUtc.SelectMany((x) => timeFunctionsUtc.SelectMany((y) => booleanArray.Select((reverse) => (x, y, reverse)))).Where((fs) => fs.x.Getter != fs.y.Getter)
```
Which can then be `Assert.All` over.
It looks horrific, which is why I had the comment above it, it could also be useful to have another comment above it with the orders that would be run when the function is run (which I was going to add), like so:
```cs
            // The order that these LINQ expression creates is (when all 3 are available):
            // [0] = (creation, access, False), [1] = (creation, access, True),  [2] = (creation, write, False),
            // [3] = (creation, write, True),   [4] = (access, creation, False), [5] = (access, creation, True),
            // [6] = (access, write, False),    [7] = (access, write, True),     [8] = (write, creation, False),
            // [9] = (write, creation, True),  [10] = (write, access, False),   [11] = (write, access, True)
            // Or, when creation time setting is not available:
            // [0] = (access, write, False),    [1] = (access, write, True),
            // [2] = (write, access, False),    [3] = (write, access, True)
```

Reply:
I have no strong opinion. We have probably spent enough time on this. 😊 @carlossanlop thoughts?

Reply:
I have reverted it to how it was before so the tests are consistent (because otherwise some will be using `Theory`, and others will be using `Assert.All`, which seems inconsistent to me when they're all just using `TimeFunctions`).
Commit: https://github.com/dotnet/runtime/pull/49555/commits/f1a303e7e1f005b5067429c94aab6ab015d1d171
It can be reverted if we decide the other way is better.
I've added/updated comments for clarity as to what it is testing, and what the indices are compared to the earlier version of this code.